test/integration/backward_compatible/parallel/serialization/compact/CompactPublicAPIsTest.js

```diff
@@ -0,0 +1,952 @@
+        this.innerField = innerField;
+    }
+}
+
+class OuterSerializer {
+    constructor() {
+        this.class = Outer;
+        this.typeName = 'Outer';
+    }
+
+    read(reader) {
+        compactSerializerUsed = true;
+        const intField = reader.readInt32('int_field');
+        const innerField = reader.readCompact('inner_field');
+        return new Outer(intField, innerField);
+    }
+
+    write(writer, instance) {
+        compactSerializerUsed = true;
+        writer.writeInt32('int_field', instance.intField);
+        writer.writeCompact('inner_field', instance.innerField);
+    }
+}
+
+class CompactPagingPredicate {
+    constructor() {
+        this.delecate = Predicates.paging(Predicates.alwaysTrue(), 1);
+    }
+
+    getComparator() {
+        return null;
+    }
+
+    nextPage() {
+        this.delecate.nextPage();
+    }
+
+    previousPage() {
+        this.delecate.previousPage();
+    }
+
+    getPage() {
+        this.delecate.getPage();
+    }
+
+    setPage(page) {
+        this.delecate.setPage(page);
+    }
+
+    getPageSize() {
+        this.delecate.getPageSize();
+    }
+
+    getAnchor() {
+        return this.delecate.getAnchor();
+    }
+}
+
+class CompactPagingPredicateSerializer {
+    constructor() {
+        this.class = CompactPagingPredicate;
+        this.typeName = 'CompactPagingPredicate';
+    }
+
+    read() {
+        compactSerializerUsed = true;
+        return new CompactPagingPredicate();
+    }
+
+    write() {
+        compactSerializerUsed = true;
+    }
+}
+
+class CompactReturningEntryProcessor {
+}
+
+class CompactReturningEntryProcessorSerializer {
+    constructor() {
+        this.class = CompactReturningEntryProcessor;
+        this.typeName = 'CompactReturningEntryProcessor';
+    }
+
+    read() {
+        compactSerializerUsed = true;
+        return new CompactReturningEntryProcessor();
+    }
+
+    write() {
+        compactSerializerUsed = true;
+    }
+}
+
+/**
+ * Tests all Public APIs if they can support compact objects.
+ */
+describe('CompactPublicAPIsTest', function () {
+    let cluster;
+    let member;
+    let client;
+    let map, multimap, list, atomicReference,
+        queue, set, topic, nearCachedMap1, nearCachedMap2,
+        replicatedMap, ringBuffer;
+    let employee;
+    let SchemaNotReplicatedError;
+    let CompactUtil;
+    let clientConfig;
+
+    const CLUSTER_CONFIG_XML = `
+        <hazelcast xmlns="http://www.hazelcast.com/schema/config"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://www.hazelcast.com/schema/config
+            http://www.hazelcast.com/schema/config/hazelcast-config-5.0.xsd">
+            <network>
+                <port>0</port>
+            </network>
+            <serialization>
+                <compact-serialization enabled="true">
+                    <registered-classes>
+                        <class type-name="${Inner.name}"
```

Comment:
does `name` defined on these classes?

Reply:
name is coming from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name

Every class is a function